api/service/legacysync/syncing.go

```diff
@@ -822,16 +848,19 @@ func (ss *StateSync) generateNewState(bc *core.BlockChain, worker *worker.Worker
 	parentHash := bc.CurrentBlock().Hash()
 
 	var err error
-	for {
-		block := ss.getBlockFromOldBlocksByParentHash(parentHash)
+	for block := ss.getBlockFromOldBlocksByParentHash(parentHash); ; {
 		if block == nil {
 			break
 		}
-		err = ss.UpdateBlockAndStatus(block, bc, worker, false)
+		next := ss.getBlockFromOldBlocksByParentHash(block.Hash())
+		// Enforce sig check for the last block in a batch
```

Comment:
why does this mean it's last block? Also can you explain the logic that involves getBlockFromOldBlocksByParentHash? what is oldBlocks?

Reply:
`oldBlocks` is just some legacy function names. oldBlocks (a.k.a. `ss.commonBlocks`) are blocks downloaded from `ss.downloadBlocks`, which makes up the major portion of synced blocks. E.g. For an out of sync node, one sync loop syncs 1000 blocks, and all these 1000 blocks downloaded from remote nodes resides in `ss.commonBlocks`.

This part of logic change is basically aiming to enforce signature verification (also flush commitSig to db) for the last block of downloaded blocks. E.g, A sync loop has downloaded 900 blocks from remote nodes, and 900th block is the latest block. So this block's sig should be verified and inserted into db, which enables consensus to find the commit sig of current block block.

Reply:
I did minimum code change but correct the logics since this is the code about to discard. Do you think I shall refactor it in a more readable way?

Reply:
got it thanks